cpp/include/cudf_test/stream_checking_resource_adaptor.hpp

```diff
@@ -73,11 +73,7 @@ class stream_checking_resource_adaptor final : public rmm::mr::device_memory_res
    *
    * @return Pointer to the upstream resource.
    */
-  [[deprecated("Use get_upstream_resource instead")]] [[nodiscard]] Upstream* get_upstream()
```

Comment:
In the cudf case this is only used in tests. As long as all the tests are updated to not call this method it's perfectly fine to deprecate, and probably even to remove it without deprecation (since it's internal to libcudf unit tests).